### ISSUE TITLE
Play completion block called before animation has finished

### ISSUE THREAD

**Issue by vbklv** [airbnb/lottie-ios#700]
## Check these before submitting:
- [x] Updated to the latest version of Lottie (2.5.0)
- [] The issue doesn't involve an [Unsupported Feature](https://github.com/airbnb/lottie-ios/blob/master/README.md#currently-unsupported-after-effects-features)
- [] This issue isn't related to another open issue

## This issue is a:
- [X] Non-Crashing Bug (Visual or otherwise)
- [] Crashing Bug
- [] Feature Request
- [] Regression (Something that once worked, but doesn't work anymore)

## What Platform are you on? 

- [] MacOS
- [X] iOS

## Expected Behavior 

The completion block is called only after animation has finished.

## Actual Behavior

The completion block is called before animation has finished.

## Code Example

		let animationView = LOTAnimationView(name: "animationName")

		animationView.loopAnimation = false

		animationContainer.addSubview(animationView)

		animationView.play { finished in
			self.addTextMessage(text: textMessage) // <----- **Executed before animation has finished**
		}

## Animation JSON

{"v":"5.2.1","fr":60,"ip":0,"op":24,"w":150,"h":150,"nm":"Error150x150","ddd":0,"assets":[{"id":"comp_0","layers":[{"ddd":0,"ind":5,"ty":4,"nm":"Error_Soft3/CheckMark_Error_icons Outlines 2","sr":1,"ks":{"o":{"a":1,"k":[{"i":{"x":[0.667],"y":[1]},"o":{"x":[0.333],"y":[0]},"n":["0p667_1_0p333_0"],"t":89,"s":[0],"e":[100]},{"t":100}],"ix":11},"r":{"a":0,"k":0,"ix":10},"p":{"a":0,"k":[153,150,0],"ix":2},"a":{"a":0,"k":[113,113,0],"ix":1},"s":{"a":1,"k":[{"i":{"x":[0.667,0.667,0.667],"y":[1,1,1]},"o":{"x":[0.333,0.333,0.333],"y":[0,0,0]},"n":["0p667_1_0p333_0","0p667_1_0p333_0","0p667_1_0p333_0"],"t":89,"s":[0,0,100],"e":[115,115,100]},{"i":{"x":[0.667,0.667,0.667],"y":[1,1,1]},"o":{"x":[0.333,0.333,0.333],"y":[0,0,0]},"n":["0p667_1_0p333_0","0p667_1_0p333_0","0p667_1_0p333_0"],"t":103,"s":[115,115,100],"e":[100,100,100]},{"t":113}],"ix":6}},"ao":0,"shapes":[{"ty":"gr","it":[{"ind":0,"ty":"sh","ix":1,"ks":{"a":0,"k":{"i":[[8.85,8.85],[0,0],[-8.85,8.85],[0,0],[-8.849,-8.849],[8.85,-8.85],[0,0]],"o":[[0,0],[-8.85,-8.85],[0,0],[8.85,-8.849],[8.85,8.85],[0,0],[-8.849,8.85]],"v":[[-103.58,103.579],[-103.58,103.579],[-103.58,71.4],[71.399,-103.581],[103.58,-103.581],[103.58,-71.4],[-71.4,103.579]],"c":true},"ix":2},"nm":"Path 1","mn":"ADBE Vector Shape - Group","hd":false},{"ty":"fl","c":{"a":0,"k":[0.933000033509,0.447000002394,0.470999983245,1],"ix":4},"o":{"a":0,"k":100,"ix":5},"r":1,"nm":"Fill 1","mn":"ADBE Vector Graphic - Fill","hd":false},{"ty":"tr","p":{"a":0,"k":[112.68,112.68],"ix":2},"a":{"a":0,"k":[0,0],"ix":1},"s":{"a":0,"k":[100,100],"ix":3},"r":{"a":0,"k":0,"ix":6},"o":{"a":0,"k":100,"ix":7},"sk":{"a":0,"k":0,"ix":4},"sa":{"a":0,"k":0,"ix":5},"nm":"Transform"}],"nm":"Group 1","np":2,"cix":2,"ix":1,"mn":"ADBE Vector Group","hd":false},{"ty":"gr","it":[{"ind":0,"ty":"sh","ix":1,"ks":{"a":0,"k":{"i":[[8.85,-8.85],[0,0],[8.85,8.85],[0,0],[-8.85,8.85],[-8.85,-8.849],[0,0]],"o":[[0,0],[-8.85,8.85],[0,0],[-8.85,-8.85],[8.849,-8.849],[0,0],[8.85,8.85]],"v":[[103.58,103.579],[103.58,103.579],[71.399,103.579],[-103.58,-71.4],[-103.58,-103.581],[-71.4,-103.581],[103.58,71.4]],"c":true},"ix":2},"nm":"Path 1","mn":"ADBE Vector Shape - Group","hd":false},{"ty":"fl","c":{"a":0,"k":[0.933000033509,0.447000002394,0.470999983245,1],"ix":4},"o":{"a":0,"k":100,"ix":5},"r":1,"nm":"Fill 1","mn":"ADBE Vector Graphic - Fill","hd":false},{"ty":"tr","p":{"a":0,"k":[112.68,112.68],"ix":2},"a":{"a":0,"k":[0,0],"ix":1},"s":{"a":0,"k":[100,100],"ix":3},"r":{"a":0,"k":0,"ix":6},"o":{"a":0,"k":100,"ix":7},"sk":{"a":0,"k":0,"ix":4},"sa":{"a":0,"k":0,"ix":5},"nm":"Transform"}],"nm":"Group 2","np":2,"cix":2,"ix":2,"mn":"ADBE Vector Group","hd":false}],"ip":89,"op":1889,"st":89,"bm":0}]}],"layers":[{"ddd":0,"ind":1,"ty":0,"nm":"Error","refId":"comp_0","sr":1,"ks":{"o":{"a":0,"k":100,"ix":11},"r":{"a":0,"k":0,"ix":10},"p":{"a":0,"k":[73.25,75.25,0],"ix":2},"a":{"a":0,"k":[150,150,0],"ix":1},"s":{"a":0,"k":[53.333,53.333,100],"ix":6}},"ao":0,"w":300,"h":300,"ip":0,"op":1710,"st":-90,"bm":0}],"markers":[]}

**Comment by buba447** [comment]
Lottie has been completely rewritten in Swift as of 3.0 (https://github.com/airbnb/lottie-ios/pull/777)

I am closing all issues prior to this release to reduce the noise. If you continue to run into this issues or any issue with Lottie 3.0 please open a **new ticket**

For continued support of Lottie Objective-c please point to this branch: https://github.com/airbnb/lottie-ios/tree/lottie/objectiveC
